src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2960,6 +2960,65 @@ public static byte[] FromHexString(ReadOnlySpan<char> chars)
             return result;
         }
 
+        /// <summary>
+        /// Converts the string, which encodes binary data as hex characters, to an equivalent 8-bit unsigned integer span.
+        /// </summary>
+        /// <param name="source">The string to convert.</param>
+        /// <param name="destination">
+        /// The span in which to write the converted 8-bit unsigned integers. When this method returns false,
+        /// either the span remains unmodified or contains an incomplete conversion of <paramref name="source"/>,
+        /// up to the last valid character.
+        /// </param>
+        /// <param name="bytesWritten">When this method returns, contains the number of bytes that were written in <paramref name="destination"/>.</param>
+        /// <returns>true if the conversion was successful; otherwise, false.</returns>
+        /// <exception cref="ArgumentNullException">Passed string <paramref name="source"/> is null.</exception>
+        public static bool TryFromHexString(string source, Span<byte> destination, out int bytesWritten)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+
+            return TryFromHexString(source.AsSpan(), destination, out bytesWritten);
+        }
+
+        /// <summary>
+        /// Converts the span of chars, which encodes binary data as hex characters, to an equivalent 8-bit unsigned integer span.
+        /// </summary>
+        /// <param name="source">The span to convert.</param>
+        /// <param name="destination">
+        /// The span in which to write the converted 8-bit unsigned integers. When this method returns false,
+        /// either the span remains unmodified or contains an incomplete conversion of <paramref name="source"/>,
+        /// up to the last valid character.
+        /// </param>
+        /// <param name="bytesWritten">When this method returns, contains the number of bytes that were written in <paramref name="destination"/>.</param>
+        /// <returns>true if the conversion was successful; otherwise, false.</returns>
+        public static bool TryFromHexString(ReadOnlySpan<char> source, Span<byte> destination, out int bytesWritten)
+        {
+            int length = source.Length;
+
+            if (length % 2 != 0)
+                goto FalseResult;
+
+            if (length == 0)
+            {
+                bytesWritten = 0;
+                return true;
+            }
+
+            int requiredByteCount = length / 2;
```

Comment:
```suggestion
            int requiredByteCount = (int)((uint)length / 2);
```

The JIT doesn't recognize `length` as never negative otherwise.

@EgorBo Should we open a JIT issue?